src/libANGLE/renderer/d3d/d3d11/Clear11.cpp

```diff
@@ -505,7 +505,34 @@ angle::Result Clear11::clearFramebuffer(const gl::Context *context,
         const auto &framebufferRTV = renderTarget->getRenderTargetView();
         ASSERT(framebufferRTV.valid());
 
-        if ((!(mRenderer->getRenderer11DeviceCaps().supportsClearView) && needScissoredClear) ||
+        bool clearViewBroken = false;
+        if (mRenderer->getFeatures().brokenClearViewAfterDualSource.enabled) {
+	    // Check the current state to see if we were using dual source blending
+            auto isDualSource = [](auto blend) { switch (blend) {
+                        case D3D11_BLEND_SRC1_COLOR:
+                        case D3D11_BLEND_INV_SRC1_COLOR:
+                        case D3D11_BLEND_SRC1_ALPHA:
+                        case D3D11_BLEND_INV_SRC1_ALPHA:
+                                return true;
+                        default:
+                                return false;
+            }};
+            FLOAT blendFactor[4];
+            UINT sampleMask;
+            ID3D11BlendState *blendState;
+            D3D11_BLEND_DESC blendDesc;
+            deviceContext1->OMGetBlendState(&blendState, blendFactor, &sampleMask);
```

Comment:
is that common in Angle to query the driver for information it should know already?

Reply:
Unlikely, but I believe ANGLE just tracks its current state and not the state that it has set. i.e. if you do a glBendFunc() before doing the glClear() ANGLE won't know what state it last drew with.